netbox_cmdb/netbox_cmdb/api/cmdb/views.py

```diff
@@ -0,0 +1,57 @@
+from django.db import transaction
+from django.db.models import Q
+from drf_yasg import openapi
+from drf_yasg.openapi import Parameter
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import serializers, status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from netbox.api.authentication import IsAuthenticatedOrLoginNotRequired
+from netbox_cmdb.models.bgp import BGPPeerGroup, BGPSession, DeviceBGPSession
+from netbox_cmdb.models.prefix_list import PrefixList
+from netbox_cmdb.models.route_policy import RoutePolicy
+from netbox_cmdb.models.snmp import SNMP
+
+
+class DeleteAllCMDBObjectsRelatedToDeviceSerializer(serializers.Serializer):
+    device_name = serializers.CharField()
+
+
+class DeleteAllCMDBObjectsRelatedToDevice(APIView):
+
+    permission_classes = [IsAuthenticatedOrLoginNotRequired]
+
+    @swagger_auto_schema(
+        request_body=DeleteAllCMDBObjectsRelatedToDeviceSerializer,
+        responses={
+            status.HTTP_200_OK: "Objects related to device have been deleted successfully",
+            status.HTTP_400_BAD_REQUEST: "Bad Request: Device name is required",
+            status.HTTP_500_INTERNAL_SERVER_ERROR: "Internal Server Error: Something went wrong on the server",
+        },
+    )
+    def post(self, request):
+        device_name = request.data.get("device_name", None)
+        if device_name is None:
+            return Response(
+                {"error": "Device name is required"}, status=status.HTTP_400_BAD_REQUEST
+            )
+
+        with transaction.atomic():
+            try:
+                # Delete objects in reverse order of dependencies
+                BGPSession.objects.filter(
+                    Q(peer_a__device__name=device_name) | Q(peer_b__device__name=device_name)
+                ).delete()
+                DeviceBGPSession.objects.filter(device__name=device_name).delete()
```

Comment:
nitpick: deleting BGPSession already deletes the attached DeviceBGPSession automatically.

Reply:
but I am ok to keep it just in cas :)

Reply:
yes that's in case there is a device BGP session that's still here for some reason (that was not attached to a BGP session for instance), that way we're sure there is no object left.